### PR TITLE
Add OP 35 documentation

### PR DESCRIPTION
**Pull request by arHSM**
Based on the following RE-d code
```ts
interface State {
    pageSize: number;
    currentPage: number;
    continuationToken: string | null;
}

interface PartialGuildMembersChunk {
    members: { joined_at: string; user: { id: string } }[];
}

const searchChunkSize = 1e3;

function getMemberCount(guildId: string): number | undefined {
    return; /* get count */
}

function getSortedMemberIDs(
    members: { joinedAtTimestamp: Date; id: string }[]
) {
    return members
        .sort(member => -member.joinedAtTimestamp)
        .map(member => member.id);
}

function calculateNewContinuationToken(
    guildId: string,
    sortedMemberIDs: string[],
    state: State
) {
    const memberCount = getMemberCount(guildId) ?? sortedMemberIDs.length;

    const sortedMemberSize = sortedMemberIDs.length - 1;
    const lastMemberID = sortedMemberIDs[sortedMemberSize];

    const minimumChunkSize = Math.min(searchChunkSize, sortedMemberSize);

    const currentIndex = state.currentPage * state.pageSize;
    const newContinuationToken =
        sortedMemberIDs[currentIndex - 1] ?? lastMemberID;

    if (null === state.continuationToken) {
        return currentIndex < minimumChunkSize
            ? state.continuationToken
            : newContinuationToken;
    }

    const previousContinuationTokenIndex = sortedMemberIDs.findIndex(
        id => id === state.continuationToken
    );

    if (-1 === previousContinuationTokenIndex) {
        return newContinuationToken;
    }

    if (currentIndex > memberCount) {
        return lastMemberID;
    }

    if (previousContinuationTokenIndex - currentIndex >= 0) {
        return state.continuationToken;
    }

    if (
        Math.ceil(currentIndex / minimumChunkSize) >
        Math.ceil(previousContinuationTokenIndex / minimumChunkSize)
    ) {
        return newContinuationToken;
    }

    return state.continuationToken;
}

function main() {
    const guildId = "1029315212005888060";
    const state: State = {
        pageSize: 50,
        currentPage: 1,
        continuationToken: null,
    };
    let sortedMemberIDs = [];

    const query = ""; // username or ID

    let exhausted = false;

    while (!exhausted) {
        socket.send(35, {
            guild_id: guildId,
            query,
            continuation_token: state.continuationToken,
        });

        socket.onEvent(
            "GUILD_MEMBERS_CHUNK",
            (data: PartialGuildMembersChunk) => {
                sortedMemberIDs = getSortedMemberIDs(
                    data.members.map(member => ({
                        joinedAtTimestamp: new Date(member.joined_at),
                        id: member.user.id,
                    }))
                );

                if (data.members.length === 0) exhausted = true;

                state.continuationToken = calculateNewContinuationToken(
                    guildId,
                    sortedMemberIDs,
                    state
                );
            }
        );
    }
}
```

### CODE REVIEW

File: pages/topics/gateway-events.mdx [hunk 1]
```diff
@@ -81,6 +81,7 @@ Send events are Gateway events encapsulated in an [event payload](#gateway-paylo
 | [Create Speed Test](#create-speed-test)                   | Create a voice speed test                                              |
 | [Delete Speed Test](#delete-speed-test)                   | Delete a voice speed test                                              |
 | [Request Last Messages](#request-last-messages)           | Request last messages for a guild's channels                           |
+| [Search recent Members](#search-recent-members)           | Similar to OP 8, used for Member Safety.                               |
```
Comment (dolfies) [inline pages/topics/gateway-events.mdx:84]:
```suggestion
| [Search Recent Members](#search-recent-members)           | Request recently-joined members for a guild                                 |
```

File: pages/topics/gateway-events.mdx [hunk 2]
```diff
@@ -332,6 +333,82 @@ Used to request a private channel's pre-existing call data, created before the G
...
+  return state["continuation_token"]
+```
+
+###### Search Recent Members Structure
+
+| Field              | Type       | Description                                 |
+| ------------------ | ---------- | ------------------------------------------- |
+| guild_id           | snowflake  | ID of guild to search                       |
+| query              | string     | A username of ID to search for              |
+| continuation_token | ?snowflake | Token/member ID to continue pagination from |
```
Comment (dolfies) [inline pages/topics/gateway-events.mdx:398]:
```suggestion
###### Search Recent Members Structure

| Field              | Type       | Description                                 |
| ------------------ | ---------- | ------------------------------------------- |
| guild_id           | snowflake  | The ID of the guild to search members for |
| query              | string     | String that the username/nickname starts with, or an empty string to return all members |
| continuation_token ^1^  | ?snowflake | The calculated token to continue pagination from |

^1^ The `continuation_token` is a member id calculated by the following Python pseudocode:

```py
from math import ceil
from typing import TypedDict

SEARCH_CHUNK_SIZE = 1000

class State(TypedDict):
  page_size: int
  current_page: int
  continuation_token: str | None

def calculate_new_continuation_token(guild_id: str, sorted_member_ids: list[str], state: State) -> str | None:
  # `sorted_member_ids` is an array of member IDs sorted by their `joined_at` field (descending)
  # You do not need to calculate a `continuation_token` if you haven't performed an initial Search Recent Members call,
  # the array is empty until a `GUILD_MEMBERS_CHUNK` event received after a Search Recent Members call,
  # sorting can be performed by converting the `joined_at` ISO8601 timestamp in the guild member object
  # to a UNIX timestamp and sorting the result in descending numerical order

  member_count = get_guild_member_count(guild_id) or len(sorted_member_ids)

  sorted_member_size = len(sorted_member_ids) - 1
  last_member_id = sorted_member_ids[sorted_member_size]

  minimum_chunk_size = min(search_chunk_size, sorted_member_size)

  current_index = state["current_page"] * state["page_size"]
  new_continuation_token = sorted_member_ids[current_index - 1] or last_member_id

  if not state["continuation_token"]:
    return state["continuation_token"] if current_index < minimum_chunk_size else new_continuation_token

  try:
    previous_continuation_token_index = sorted_member_ids.index(state["continuation_token"])

    if current_index > member_count:
      return last_member_id

    if (previous_continuation_token_index - current_index) >= 0:
      return state["continuation_token"]

    if ceil(current_index / minimum_chunk_size) > ceil(previous_continuation_token_index / minimum_chunk_size):
      return new_continuation_token
  except ValueError:
    return new_continuation_token

  return state["continuation_token"]
```
```
Is `continuation_token` optional?

File: pages/topics/gateway-events.mdx [hunk 3]
```diff
@@ -332,6 +333,82 @@ Used to request a private channel's pre-existing call data, created before the G
 }
 ```
 
+#### Search Recent Members
+
+{/* TODO: improve description */}
+
+Used to search members for Member Safety. You can request up-to 1000 members per call
```
Comment (dolfies) [inline pages/topics/gateway-events.mdx:340]:
```suggestion
Used to search the recently joined members in a guild. Fires a [Guild Members Chunk](#guild-members-chunk) Gateway event with up to 1000 members that match the request.
```

File: pages/topics/gateway-events.mdx [hunk 4]
```diff
@@ -332,6 +333,82 @@ Used to request a private channel's pre-existing call data, created before the G
...
+
+```json
+{
+  "op": 35,
+  "d": {
+    "guild_id": "957057010334048288",
+    "query": ""
+  }
+}
+```
```
Comment (dolfies) [inline pages/topics/gateway-events.mdx:410]:
```suggestion
```json
{
  "op": 35,
  "d": {
    "guild_id": "957057010334048288",
    "query": "",
    "continuation_token": null
  }
}
```
```

**Review by dolfies — CHANGES_REQUESTED** [review]

File: pages/topics/gateway-events.mdx (hunk 2 above)
Comment (arHSM) [reply pages/topics/gateway-events.mdx:398]: No its not optional, but it is nullable

File: pages/topics/gateway-events.mdx [hunk 5]
```diff
@@ -332,6 +333,80 @@ Used to request a private channel's pre-existing call data, created before the G
...
+
+    if (previous_continuation_token_index - current_index) >= 0:
+      return state["continuation_token"]
+
+    if ceil(current_index / minimum_chunk_size) > ceil(previous_continuation_token_index / minimum_chunk_size):
+      return new_continuation_token
+  except ValueError:
+    return new_continuation_token
+
+  return state["continuation_token"]
```
Comment (dolfies) [inline pages/topics/gateway-events.mdx:396]:
```suggestion
| Field                   | Type       | Description                                 |
| ----------------------- | ---------- | ------------------------------------------- |
| guild_id                | snowflake  | The ID of the guild to search members for |
| query ^1^               | string     | String that the username/nickname starts with, or an empty string to return all members |
| continuation_token ^2^  | ?snowflake | The member ID to continue pagination from |

^1^ When a query is provided, results are limited to one member.

^2^ Pagination is based on the [`joined_at`](/resources/guild#guild-member-object) field descending. To paginate, you provide the member ID of the member that joined the earliest in the previously received [Guild Members Chunk](#guild-members-chunk).

```
sorry!

**Review by dolfies — APPROVED** [review]

**Pull request merged**